### PR TITLE
modules/SceRegistryMgr: Intial implement of regmgr funcs.

### PR DESCRIPTION
**Pull request by Zangetsu38**
# About:
- Intial implement of regmgr.

# Result
- on registry editor
image
image

### CODE REVIEW

**Comment by Macdu** [comment]
Shouldn't the RegMngr functions be thread safe?

File: vita3k/modules/SceRegistryMgr/SceRegMgr.cpp [hunk 1]
```diff
@@ -40,30 +42,34 @@ EXPORT(int, sceRegMgrGetInitVals) {
...
+
+    return 0;
 }
 
-EXPORT(int, sceRegMgrGetKeyInt, const char *category, const char *name, int *buf) {
+EXPORT(int, sceRegMgrGetKeyInt, const char *category, const char *name, SceInt32 *buf) {
     TRACY_FUNC(sceRegMgrGetKeyInt, category, name, buf);
-    if ((std::string(category) == "/CONFIG/PSPEMU") && (std::string(name) == "emu_list_flag")) {
-        STUBBED("Stubbed");
-        *buf = 1;
```
Comment (Macdu) [inline vita3k/modules/SceRegistryMgr/SceRegMgr.cpp:52]: What was this used for? Will it break some game?

File: vita3k/regmgr/src/regmgr.cpp [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+// 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+#include <regex>
+#include <regmgr/functions.h>
+
+#include <util/log.h>
+
+namespace regmgr {
+
+const std::vector<unsigned char> xorKey = { 0x89, 0xFA, 0x95, 0x48, 0xCB, 0x6D, 0x77, 0x9D, 0xA2, 0x25, 0x34, 0xFD, 0xA9, 0x35, 0x59, 0x6E };
```
Comment (Macdu) [inline vita3k/regmgr/src/regmgr.cpp:25]: use a std::array instead

File: vita3k/regmgr/src/regmgr.cpp [hunk 3]
```diff
@@ -0,0 +1,361 @@
...
+            continue;
+
+        // Found the base register
+        if (line.find("[BASE") != std::string::npos) {
+            // Register the names with their numbers
+            while (std::getline(iss, line) && (line != "[REG-BAS")) {
+                if (line.empty())
+                    continue;
+
+                std::size_t pos = line.find('=');
```
Comment (Macdu) [inline vita3k/regmgr/src/regmgr.cpp:78]:
```suggestion
                size_t pos = line.find('=');
```
and same for the others

File: vita3k/modules/SceRegistryMgr/SceRegMgr.cpp (hunk 1 above)
Comment (Zangetsu38) [reply vita3k/modules/SceRegistryMgr/SceRegMgr.cpp:52]: nop, is my old stub, now i using real value

File: vita3k/regmgr/src/regmgr.cpp (hunk 3 above)
Comment (Zangetsu38) [reply vita3k/regmgr/src/regmgr.cpp:78]: ah yes

Comment (Zangetsu38) [reply vita3k/regmgr/src/regmgr.cpp]: done

File: vita3k/regmgr/src/regmgr.cpp (hunk 2 above)
Comment (Zangetsu38) [reply vita3k/regmgr/src/regmgr.cpp]: done

File: vita3k/modules/SceRegistryMgr/SceRegMgr.cpp (hunk 1 above)
Comment (Macdu) [reply vita3k/modules/SceRegistryMgr/SceRegMgr.cpp:52]: Well I was wondering since there is no /CONFIG/PSPEMU in get_category_and_name_by_id

Comment (Zangetsu38) [reply vita3k/modules/SceRegistryMgr/SceRegMgr.cpp:52]: normal, module regmgr no using id, it give directly category and name
is module  ForGame and ForSdk using id
and for now, i have register in this function all id i have see called
i have add log, for can adding next in futur is view

**Review by Macdu — APPROVED** [review]
lgtm, although I don't think unordered_map is the best data structure to use here

**Pull request merged**